api/tests/test-router.js

```diff
@@ -239,94 +239,120 @@ o.spec("route", function() {
 							return m("div")
 						}
 					}
-					
+
 					$window.location.href = prefix + "/"
 					route(root, "/abc", {
 						"/:id" : {
 							onmatch: function(vnode, resolve) {
 								matchCount++
-								
+
 								o(vnode.attrs.id).equals("abc")
 								o(vnode.attrs.path).equals("/abc")
 								o(vnode.attrs.route).equals("/:id")
-								
+
 								resolve(Component)
 							},
 							view: function(vnode) {
 								renderCount++
-								
+
 								o(vnode.attrs.id).equals("abc")
-								
+
 								return vnode
 							},
 						},
 					})
-					
+
 					setTimeout(function() {
 						o(matchCount).equals(1)
 						o(renderCount).equals(1)
 						o(root.firstChild.nodeName).equals("DIV")
-						
+
 						done()
 					}, FRAME_BUDGET)
 				})
-				
+
+				o("onmatch can redirect to another route", function(done) {
+					var redirected = false
+
+					$window.location.href = prefix + "/"
+					route(root, "/a", {
+						"/a" : {
+							onmatch: function() {
+								route.set("/b")
+							}
+						},
+						"/b" : {
+							view: function(vnode){
+								redirected = true
+
+								return vnode
```

Comment:
This ties back into my argument that the distinction between RouteResolvers & Components as expounded in that long thread (in particular WRT [Leo's opinion that `RouteResolver.render` be considered as a different kind of thing to `Component.view`](https://github.com/lhorie/mithril.js/issues/1095#issuecomment-240726967) is actually very muddled, and the separation of concerns is screwed according to an idea of an API that doesn't actually make author-land sense when compared to the documentation and the purported benefits. 

Things are currently broken either way. Let's deal with your PR in its own thread ;)